config/notifiers.go

```diff
@@ -870,6 +882,77 @@ func (c *JiraConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+}
+
+type RocketchatAttachmentActionType string
+
+type RocketchatMessageProcessingType string
+
+const (
+	ProcessingTypeSendMessage        RocketchatMessageProcessingType = "sendMessage"
+	ProcessingTypeRespondWithMessage RocketchatMessageProcessingType = "respondWithMessage"
+)
+
+type RocketchatAttachmentAction struct {
+	Type               RocketchatAttachmentActionType  `json:"type,omitempty"`
+	Text               string                          `json:"text,omitempty"`
+	URL                string                          `json:"url,omitempty"`
+	ImageURL           string                          `json:"image_url,omitempty"`
+	IsWebView          bool                            `json:"is_webview"`
+	WebviewHeightRatio string                          `json:"webview_height_ratio,omitempty"`
+	Msg                string                          `json:"msg,omitempty"`
+	MsgInChatWindow    bool                            `json:"msg_in_chat_window"`
+	MsgProcessingType  RocketchatMessageProcessingType `json:"msg_processing_type,omitempty"`
+}
 
+// RocketchatConfig configures notifications via Rocketchat.
+type RocketchatConfig struct {
+	NotifierConfig `yaml:",inline" json:",inline"`
+
+	HTTPConfig *commoncfg.HTTPClientConfig `yaml:"http_config,omitempty" json:"http_config,omitempty"`
+
+	APIURL      *URL    `yaml:"api_url,omitempty" json:"api_url,omitempty"`
+	TokenID     *Secret `yaml:"token_id,omitempty" json:"token_id,omitempty"`
+	TokenIDFile string  `yaml:"token_id_file,omitempty" json:"token_id_file,omitempty"`
+	Token       *Secret `yaml:"token,omitempty" json:"token,omitempty"`
+	TokenFile   string  `yaml:"token_file,omitempty" json:"token_file,omitempty"`
+
+	// RocketChat channel override, (like #other-channel or @username).
+	Channel string `yaml:"channel,omitempty" json:"channel,omitempty"`
+
+	Color       string                        `yaml:"color,omitempty" json:"color,omitempty"`
+	Title       string                        `yaml:"title,omitempty" json:"title,omitempty"`
+	TitleLink   string                        `yaml:"title_link,omitempty" json:"title_link,omitempty"`
+	Text        string                        `yaml:"text,omitempty" json:"text,omitempty"`
+	Fields      []*RocketchatAttachmentField  `yaml:"fields,omitempty" json:"fields,omitempty"`
+	ShortFields bool                          `yaml:"short_fields" json:"short_fields,omitempty"`
+	Emoji       string                        `yaml:"emoji,omitempty" json:"emoji,omitempty"`
+	IconURL     string                        `yaml:"icon_url,omitempty" json:"icon_url,omitempty"`
+	ImageURL    string                        `yaml:"image_url,omitempty" json:"image_url,omitempty"`
+	ThumbURL    string                        `yaml:"thumb_url,omitempty" json:"thumb_url,omitempty"`
+	LinkNames   bool                          `yaml:"link_names" json:"link_names,omitempty"`
+	Actions     []*RocketchatAttachmentAction `yaml:"actions,omitempty" json:"actions,omitempty"`
```

Comment:
Do we also need a validation rule on the maximum number of actions?

Reply:
Same as above. Also, what would be a sensible limit?